ext/dom/document.c

```diff
@@ -1660,17 +1619,19 @@ static void dom_xinclude_strip_references(xmlNodePtr basep)
 	}
 }
 
-/* See GH-14702.
- * We have to remove userland references to xinclude fallback nodes because libxml2 will make clones of these
+/* See GH-14702 and GH-17847.
+ * We have to remove userland references to xinclude nodes because libxml2 will make clones of these
  * and remove the original nodes. If the originals are removed while there are still userland references
  * this will cause memory corruption. */
-static void dom_xinclude_strip_fallback_references(const xmlNode *basep)
+static void dom_xinclude_strip_fallback_references(xmlDocPtr docp, const xmlNode *basep)
```

Comment:
Why is the docPtr necessary here?

Reply:
Leftover from when I was playing around with this. Dropped.